lib/Transforms/ConvertSecretExtractToStaticExtract/ConvertSecretExtractToStaticExtract.cpp

```diff
@@ -0,0 +1,212 @@
+#include "mlir/include/mlir/Dialect/Affine/IR/AffineOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"    // from @llvm-project
+#include "mlir/include/mlir/Dialect/SCF/IR/SCF.h"        // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tensor/IR/Tensor.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Diagnostics.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"   // from @llvm-project
+#include "mlir/include/mlir/IR/MLIRContext.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"              // from @llvm-project
+#include "mlir/include/mlir/IR/PatternMatch.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"                  // from @llvm-project
+#include "mlir/include/mlir/Interfaces/SideEffectInterfaces.h"  // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"           // from @llvm-project
+#include "mlir/include/mlir/Support/LogicalResult.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+
+namespace mlir {
+namespace heir {
+
+#define GEN_PASS_DEF_CONVERTSECRETEXTRACTTOSTATICEXTRACT
+#include "lib/Transforms/ConvertSecretExtractToStaticExtract/ConvertSecretExtractToStaticExtract.h.inc"
+
+struct SecretExtractToStaticExtractConversion
+    : OpRewritePattern<tensor::ExtractOp> {
+  using OpRewritePattern<tensor::ExtractOp>::OpRewritePattern;
+
+ public:
+  SecretExtractToStaticExtractConversion(DataFlowSolver *solver,
+                                         MLIRContext *context)
+      : OpRewritePattern(context), solver(solver) {}
+
+  LogicalResult matchAndRewrite(tensor::ExtractOp extractOp,
+                                PatternRewriter &rewriter) const override {
+    // TODO(#866): Add support for reads on multi-dimensional tensors
+    if (extractOp.getTensor().getType().getRank() != 1) {
+      extractOp->emitWarning()
+          << "Currently, transformation only supports 1D tensors:";
+      return failure();
+    }
+
+    auto index = extractOp.getIndices().front();
+
+    auto *indexSecretnessLattice =
+        solver->lookupState<SecretnessLattice>(index);
+
+    // If (1) lattice is not found or (2) lattice is set to unknown, apply
+    // transformation
+    if (!indexSecretnessLattice ||
+        !indexSecretnessLattice->getValue().isInitialized()) {
+      extractOp->emitWarning()
+          << "Secretness for tensor.extract index has not been set. "
+             "Conservatively, the transformation will be applied:";
+    }
+
+    auto indexSecretness = indexSecretnessLattice->getValue();
+
+    bool isIndexSecret =
+        !indexSecretness.isInitialized() ||
+        (indexSecretness.isInitialized() && indexSecretness.getSecretness());
+
+    // If index is not secret, no transformation is needed
+    if (!isIndexSecret) return failure();
```

Comment:
This has a few issues, most importantly it'll crash on a missing lattice due to line 64 dereferencing a null pointer ;) 
I'd suggest following the same approach as I did in fixing ConvertIfToSelect (cf. #889):


```suggestion
    // use secretness from lattice or, if no lattice found, set to unknown
    auto indexSecretness = indexSecretnessLattice ? indexSecretnessLattice->getValue() : Secretness();

    // If (1) lattice is not found or (2) lattice is set to unknown, 
    // apply transformation anyway but emit a warning
    if (!indexSecretness.isInitialized()) {
      extractOp->emitWarning()
          << "Secretness for tensor.extract index is unknown. "
             "Conservatively, the transformation will be applied:";
    }

    // If index is known to be public, no transformation is needed
    if (indexSecretness.isInitialized() && !indexSecretness.getSecretness())
        return failure();
```